internal/util/argParser.go

```diff
@@ -1,55 +1,167 @@
 package util
 
 import (
+	"encoding/json"
 	"fmt"
 	"strings"
 )
 
-// Checks and parses input parameters of type "--foo=bar" and "--foo 1 1 1 1"
-// returns fixed argument list
-func CleanInputArguments(arguments []string) ([]string, error) {
+type Input struct {
+	ParameterHierarchy []string
+	Value              string
+	Unset              bool
+	IsJson             bool
```

Comment:
This isn't really wrong, but I don't think it is really necessary to know at this point if the value is valid json or not, because even if it is valid json it may turn out that the destination field is actually a string and we don't really need to do any json parsing. I would remove the json logic from the argument parsing stage and only do it in the "applying arguments to data structures" stage.

But again, it's not wrong the way it is, so feel free to leave it as it is.

Reply:
Refactored in c1f8eec.